components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -224,9 +234,46 @@ def get_clp_connection_params_and_type(self, disable_localhost_socket_connection
         return connection_params_and_type
 
     def dump_to_primitive_dict(self):
-        d = self.model_dump(exclude={"username", "password"})
+        d = self.model_dump(
+            exclude={"username", "password", "privileged_username", "privileged_password"}
+        )
         return d
 
+    def has_root_password(self) -> bool:
+        """
+        Checks if root password is configured.
+
+        :return: True if root password is set.
+        """
+        return self.root_password is not None
+
+    def has_privileged_credentials(self) -> bool:
+        """
+        Checks if privileged credentials are configured.
+
+        :return: True if both privileged username and password are set.
+        """
+        return self.privileged_username is not None and self.privileged_password is not None
+
+    def get_privileged_connection_params(self, disable_localhost_socket_connection: bool = False):
+        """
+        Gets MySQL connection parameters using privileged credentials.
+
+        :param disable_localhost_socket_connection:
+        :return:
+        :raises ValueError: If privileged credentials are not configured.
+        """
+        if not self.has_privileged_credentials():
+            raise ValueError(
+                "Privileged database credentials not configured."
+                " Set privileged_username and privileged_password in database configuration."
+            )
```

Comment:
can we extract a `ensure_admin_credentials_loaded` method for this? (should be similar to similar to `ensure_credentials_loaded`